### ISSUE TITLE
Defining customState in Auth.federatedSignIn Changes Provider Value

### ISSUE THREAD

**Issue by sinamotamedi** [aws-amplify/amplify-js#10900]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication

### Amplify Categories

auth

### Environment information

<details>

```
# Put output below this line
System:
    OS: macOS 13.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 133.85 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.3.0 - ~/.nvm/versions/node/v18.3.0/bin/node
    npm: 8.11.0 - ~/.nvm/versions/node/v18.3.0/bin/npm
  Browsers:
    Chrome: 109.0.5414.87
    Firefox: 102.6.0
    Safari: 16.2
  npmPackages:
    @babel/core: ^7.10.3 => 7.11.1 
    @babel/preset-env: ^7.10.3 => 7.11.0 
    @babel/preset-react: ^7.10.1 => 7.10.4 
    @babel/preset-typescript: ^7.10.1 => 7.10.4 
    @dnd-kit/core: ^6.0.5 => 6.0.5 
    @dnd-kit/modifiers: ^6.0.0 => 6.0.0 
    @dnd-kit/sortable: ^7.0.1 => 7.0.1 
    @emotion/core: ^10.0.28 => 10.0.28 
    @emotion/is-prop-valid: ^0.8.8 => 0.8.8 
    @emotion/styled: ^10.0.27 => 10.0.27 
    @katal/manifest-webpack-middleware: ^1.0.0 => 1.0.0 
    @katal/manifest-webpack-plugin: ^1.0.0 => 1.0.0 
    @katal/metricsDriverSushi: ^1.0 => 1.0.0 
    @reduxjs/toolkit: ^1.7.2 => 1.7.2 
    @reduxjs/toolkit-query:  1.0.0 
    @reduxjs/toolkit-query-react:  1.0.0 
    @testing-library/jest-dom: ^5.11.0 => 5.11.2 
    @testing-library/react: ^10.4.3 => 10.4.8 
    @testing-library/user-event: ^12.1.10 => 12.8.3 
    @types/jest: ^25.1.4 => 25.2.3 (23.3.14)
    @types/react: ^16.9.41 => 16.9.45 
    @types/react-dom: ^16.9.5 => 16.9.8 
    @typescript-eslint/eslint-plugin: ^2.25.0 => 2.34.0 
    @typescript-eslint/parser: ^2.26.0 => 2.34.0 
    awesome-typescript-loader: ^5.2.1 => 5.2.1 
    aws-amplify: ^1.1.35 => 1.3.3 
    aws-amplify-react: ^2.3.12 => 2.6.3 
    axios: ^0.19.2 => 0.19.2 
    babel-jest: ^25.2.4 => 25.5.1 (26.6.3)
    babel-loader: ^8.0.6 => 8.1.0 
    babel-plugin-emotion: ^10.0.29 => 10.0.33 
    clean-webpack-plugin: ^3.0.0 => 3.0.0 
    css-loader: ^3.6.0 => 3.6.0 
    eslint: ^6.8.0 => 6.8.0 
    eslint-config-prettier: ^6.10.1 => 6.11.0 
    eslint-loader: ^3.0.3 => 3.0.4 
    eslint-plugin-import: ^2.22.0 => 2.22.0 
    eslint-plugin-react: ^7.20.1 => 7.20.5 
    eslint-plugin-react-hooks: ^3.0.0 => 3.0.0 
    file-loader: ^6.0.0 => 6.0.0 
    html-webpack-plugin: ^3.2.0 => 3.2.0 
    i18next: ^19.5.1 => 19.6.3 (19.9.2)
    i18next-http-backend: ^1.0.15 => 1.0.17 
    i18next-sprintf-postprocessor: ^0.2.2 => 0.2.2 
    image-webpack-loader: ^6.0.0 => 6.0.0 
    jest: ^26.6.3 => 26.6.3 
    jest-emotion: ^10.0.32 => 10.0.32 
    memo-parser:  0.2.1 
    mocker-api: ^2.1.0 => 2.3.0 
    react: ^16.12.0 => 16.13.1 
    react-dom: ^16.12.0 => 16.13.1 
    react-hot-loader: ^4.12.19 => 4.12.21 
    react-i18next: ^11.7.0 => 11.7.0 (11.8.10)
    react-redux: ^7.2.6 => 7.2.6 
    react-router-dom: ^5.2.0 => 5.2.0 
    react-test-renderer: ^16.13.1 => 16.13.1 
    reset-css: ^5.0.1 => 5.0.1 
    source-map-loader: ^0.2.4 => 0.2.4 
    style-loader: ^1.1.3 => 1.2.1 
    svg-url-loader: ^5.0.1 => 5.0.1 
    ts-jest: ^25.3.0 => 25.5.1 
    ts-loader: ^6.2.1 => 6.2.2 
    typescript: ^3.9.5 => 3.9.7 
    url-loader: ^3.0.0 => 3.0.0 
    webpack: ^4.41.5 => 4.44.1 
    webpack-cli: ^3.3.12 => 3.3.12 
    webpack-dev-server: ^3.10.2 => 3.11.0 
    webpack-merge: ^4.2.2 => 4.2.2 
  npmGlobalPackages:
    artillery: 2.0.0-23
    corepack: 0.10.0
    electron: 19.0.4
    expo-cli: 6.0.2
    jest: 29.0.3
    npm: 8.11.0
    typescript: 4.7.4

```

</details>

### Describe the bug

We bypass the Cognito hosted UI by defining a `customProvider` like `Auth.federatedSignIn({ customProvider: 'abc' })`. 

However, when we add `customState` to that call like `Auth.federatedSignIn({ customProvider: 'abc', customState: 'xyz' })` the user gets redirected to the hosted UI.

When inspecting the login button's onclick handler, we can see that the value we provide in `customProvider` is ignored and it defaults to `identity_provider=COGNITO` instead of `identity_provider=abc` before we supplied `customState`.

### IdP correctly populated when customState not defined
```
Auth.federatedSignIn({ customProvider: 'abc' }) =>

/authorize
redirect_uri=xxx
response_type=code
client_id=xxx
identity_provider=abc
scopes=openid
state=xxx
code_challenge=xxx
code_challenge_method=S256
```

### IdP defaults to COGNITO when customState is defined
```
Auth.federatedSignIn({ customProvider: 'abc', customState: 'xyz' }) =>

/authorize
redirect_uri=xxx
response_type=code
client_id=xxx
identity_provider=COGNITO
scopes=openid
state=xxx-xyz
code_challenge=xxx
code_challenge_method=S256
```

### Expected behavior

When `customState` is provided, the hosted UI should continue to be bypassed if `customProvider` is defined.

### Reproduction steps

1. Add `customState` to `Auth.federatedSignIn` call like:
```
Auth.federatedSignIn({ customProvider: 'abc', customState: 'xyz' })
```
2. Delete existing auth cookies/cache to force re-auth
3. Load app
4. Observe hosted UI
5. Inspect login button element's onclick handler and look for `identity_provider` value

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by cwomack** [comment]
Hello @sinamotamedi 👋.  After trying to reproduce this issue, I noticed that your version of Amplify appears to be very outdated.  There appears to be similar issues that were opened such as #4155 and #5860 (both now closed), with a fix that should have been [implemented in v3.0.23](https://github.com/aws-amplify/amplify-js/issues/5860#issuecomment-670676981).  

Can you try upgrading Amplify to the latest version and see if that resolves the issue?

**Comment by sinamotamedi** [comment]
Thanks. Can confirm updating fixed it. Closing issue.
